brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -97,21 +97,78 @@
  * }</pre>
  */
 public final class ExtraFieldPropagation<K> implements Propagation<K> {
+  static final FieldCustomizer NOOP_FIELD_CUSTOMIZER = new FieldCustomizer() {
+    @Override public String onField(String key, String value) {
+      return value;
+    }
+  };
+  static final Customizer[] NO_CONTEXT_CUSTOMIZERS = new Customizer[0];
+  static final FieldCustomizer[] NO_FIELD_CUSTOMIZERS = new FieldCustomizer[0];
+
+  /**
+   * This allows you to manipulate fields, including dropping or initializing values. This object
+   * will be called for each field added to {@link FactoryBuilder}, in the same order they were,
+   * added and regardless of whether the field is present or not.
+   */
+  public interface FieldCustomizer {
```

Comment:
I am curious about the contracts on this interface. I see this method doing two things (setting a value or dropping a value) and I would like to understand the reason for not having two methods.

Reply:
because it is easier this way, also you can implement single-method types in a lambda.

Reply:
by easier I mean there's a very direct relationship without ordering concerns about what we are doing here. it is also easier on the code to not have to enter two methods for the same outcome (ex the callsite is a loop of only one method call vs two calls one to change one to drop)